templates/seL4VMParameters.template.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2023, DornerWorks
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <camkes.h>
+#include <arm_vm/gen_config.h>
+
```

Comment:
Please create a `seL4VMParameters.template.h` with the interface, so this can be included where needed. This avoids redundant manual definitions scattered over various c files.

Reply:
Shouldn't be needed since all the variables are constants now.

Reply:
I consider it a good practice if a C file has a header with the interface. So everybody must includes the header instead of pulling in definitions directly in demand. It's better controlled then in the long run, and there is less chance different definitions exists in the code base that can break unnoticed. We have two C files already that copy the same definitions.

Reply:
Im confused. Do you want me to implement functions like this in the C source, and then replace everywhere the global variable is used in main.c?

```
int get_ram_base(void) {
    return ram_base;
}
```

Reply:
It just a standard C header file. So for any 
```
const unsigned long ram_size = /*? vm_address_config.get('ram_size') ?*/;
```
there is this in the header:
```
extern const unsigned long ram_size;
```
And whoever need something from the C file (in our case `main.c` and `init_ram.c`) has to include the header instead of directly cherry picking the extern declaration. It seem a bit overkill here, but it's worth it for the long term maintainability of the template.

Reply:
Makes sense. That's a good change. Done.